Correcting documentation of Drag/LiftedDrag (Qiskit#7659)

tobias-kehrer opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add a release note file using reno if the change needs to be
  documented in the release notes.
⚠️ If your pull request fixes an open issue, please link to the issue.

- [ ] I have added the tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the CONTRIBUTING document.
-->

### Summary
This PR partially fixes #7659.
As discussed in #7659, the documentation of the Drag/LiftedDrag pulse was not consistent with its implementation in `qiskit.pulse.library.continuous.py`.

A remaining problem is the explicit definition of the sampler strategy. This should be fixed after/with the update of parametric pulses (#7821).



### Details and comments
This PR corrects the definition of `f(x)` (called Drag/LiftedDrag, https://qiskit.org/documentation/stubs/qiskit.pulse.library.Drag.html) up to explicitly defining the chosen sampler strategy.

Before the PR, the function `f(x)` was wrongly depending on `(f'(x) - f'(-1))/(1 - f'(-1))`. This dependency is corrected to `(g(x) - g(-1))/(1 - g(-1))`. Furthermore, the imaginary part of the `f(x)` is now correctly defined to be proportional to `(g(x) - g(-1))/(1 - g(-1))`.

The documentation of the Drag/LiftedDrag pulse now follows its implementation in `qiskit.pulse.library.continuous.py`.

As @nkanazawa1989 pointed out in #7659, an update of parametric pulses (#7821) will unify the sampler strategies on backend and frontend. After or with this update, the documentation can be adjusted (if needed) stating explicitly which sampler strategy is or can be used.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=7856) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Qiskit/qiskit-terra?pullRequest=7856) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Qiskit/qiskit-terra?pullRequest=7856) it.</sub>

nkanazawa1989 reviewed (commented):
Seems like this is correct description of the function. However, since f'(x) is no longer used, we can remove another lines as well or introduce something like g'(x) to get more compact expression?

tobias-kehrer commented:
I implemented the following changes:

- Removal of f'(x)
- Added definition of g'(x)
- Typos corrected 'gaussian' -> 'Gaussian'
- Link to `qiskit.pulse.library.Gaussian` added

javabster commented:
Hi @tobias-kehrer thanks for your work! Looks like you just have some minor linting errors to fix up in the following files:

```
qiskit/pulse/library/parametric_pulses.py:353:0: C0301: Line too long (127/105) (line-too-long)
qiskit/pulse/library/parametric_pulses.py:356:0: C0301: Line too long (129/105) (line-too-long)
qiskit/pulse/library/parametric_pulses.py:358:0: C0301: Line too long (114/105) (line-too-long)
```

tobias-kehrer commented:
Thank you @javabster for pointing out these errors!
There is a remaining error

`qiskit/pulse/library/parametric_pulses.py:357:31: C0303: Trailing whitespace (trailing-whitespace)`

probably due to the 'long' link in line 358. But this link should not be split in half, right?

nkanazawa1989 commented on qiskit/pulse/library/parametric_pulses.py:
```diff
@@ -341,23 +341,24 @@ class Drag(ParametricPulse):
-    :math:`f'(x)` is the DRAG waveform without lifting or amplitude scaling.
+    where :math:`g(x)` is a standard unlifted Gaussian waveform and
+    :math:`g'(x)` is the lifted 
+    `Gaussian <https://qiskit.org/documentation/stubs/qiskit.pulse.library.Gaussian.html>`_ waveform.
```
```suggestion
    :class:`~qiskit.pulse.library.Gaussian` waveform.
```
since this page lives in the same apidocs, you can directly jump to the class api docs like this.

nkanazawa1989 reviewed (approved):
Thanks. This looks good to me.
